src/main/java/com/whiteclarkegroup/liquibaselinter/config/rules/core/AllowableXsdVersionsRule.java

```diff
@@ -0,0 +1,82 @@
+package com.whiteclarkegroup.liquibaselinter.config.rules.core;
+
+import com.google.auto.service.AutoService;
+import com.whiteclarkegroup.liquibaselinter.config.rules.AbstractLintRule;
+import com.whiteclarkegroup.liquibaselinter.config.rules.ChangeLogRule;
+import com.whiteclarkegroup.liquibaselinter.config.rules.RuleConfig;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.exception.UnexpectedLiquibaseException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+import java.io.FileReader;
+import java.util.Iterator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@AutoService({ChangeLogRule.class})
+public class AllowableXsdVersionsRule extends AbstractLintRule implements ChangeLogRule {
+    private static final Pattern DBCHANGELOG_DEFAULT_MATCH = Pattern.compile(".*dbchangelog-(.*).xsd");
+    private static final String NAME = "allowable-xsd-versions";
+    private static final String MESSAGE = "Changelog should use an XSD version in %s";
+    private static final XMLInputFactory factory = XMLInputFactory.newInstance();
+
+    private Pattern dbChangelogPattern = DBCHANGELOG_DEFAULT_MATCH;
+
+
+    public AllowableXsdVersionsRule() throws ParserConfigurationException {
+        super(NAME, MESSAGE);
+    }
+
+    @Override
+    public void configure(RuleConfig ruleConfig) {
+        super.configure(ruleConfig);
+        if (ruleConfig.hasPattern()) {
+            this.dbChangelogPattern = Pattern.compile(ruleConfig.getPatternString());
+        }
+    }
+
+    @Override
+    public boolean invalid(DatabaseChangeLog changeLog) {
+        String xsdVersion = getXsdVersion(changeLog);
+        return !ruleConfig.getValues().contains(xsdVersion);
+    }
+
+    private String getXsdVersion(DatabaseChangeLog changeLog) {
+        try (FileReader reader = new FileReader(changeLog.getPhysicalFilePath());){
+            XMLEventReader eventReader = factory.createXMLEventReader(reader);
```

Comment:
Wondering if we could use existing extension hooks to grab the file content? Even if it means a little refactoring

https://liquibase.jira.com/wiki/spaces/CONTRIB/pages/2424849/ChangeLogParser